int32 arm_type
